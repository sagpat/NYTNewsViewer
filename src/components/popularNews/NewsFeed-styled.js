import styled from 'styled-components';

export const Heading = styled.header`
    text-align: center;
    font-size: 30px;
    padding: 15px;
`;

export const ItemDiv = styled.header`
    width: 100%
`;

export const UnorderedList = styled.ul`
    list-style-type: none;
    margin: 1% 20%;
`;

export const ListedItem = styled.li`
    display: flex;
    padding: 10px;
    border-bottom: 1px solid #89ec26;
    cursor: pointer;
`;

export const HeadingLevelOne = styled.h1`
    font-size: 20px;
    font-weight: 400;
    margin-bottom: 0;
`;

export const Section = styled.section`
    display: flex;
`;


export const HeadingTwoAuthor = styled.h2`
    font-size: 16px;
    font-weight: 400;
`;

export const HeadingTwoDate = styled.h2`
    font-size: 16px;
    font-weight: 400;
    margin-left: auto;
    order: 2;
`;

export const Button = styled.button`
    padding-top: 2%;
    padding-left: 20px;
    color: red;
    font-size: 16px;
    font-weight: bold;
    border: none;
    background: none;
    outline: none;
    cursor: pointer;
`;