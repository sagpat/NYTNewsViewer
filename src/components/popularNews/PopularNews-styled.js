import styled from 'styled-components';

export const Heading = styled.header`
    text-align: center;
    font-size: 30px;
    padding: 15px;
`;

export const ItemDiv = styled.div`
    width: 100%
`;

export const UnorderedList = styled.ul`
    list-style-type: none;
    margin: 1% 20%;
`;

export const ListedItem = styled.li`
    display: flex;
    padding: 10px;
    border-bottom: 1px solid #89ec26;
`;

export const Image = styled.img`
    margin-right: 15px;
    border-radius: 50%;
`

export const TitleButton = styled.button`
    font-size: 17px;
    font-weight: 400;
    padding-left: 0;
    margin-bottom: 0;
    text-align: left;
    border: none;
    background: none;
    outline: none;
    cursor: pointer;
`;

export const SummaryParagraph = styled.p`
    padding-right: 5px;
`;

export const Section = styled.section`
    display: flex;
`;


export const HeadingTwoAuthor = styled.h2`
    font-size: 16px;
    font-weight: 400;
`;

export const HeadingTwoDate = styled.h2`
    font-size: 16px;
    font-weight: 400;
    margin-left: auto;
    order: 2;
`;

export const Button = styled.button`
    padding-top: 2%;
    padding-left: 20px;
    color: red;
    font-size: 16px;
    font-weight: bold;
    border: none;
    background: none;
    outline: none;
    cursor: pointer;
`;

export const NewsSection = styled.div`
    border: 1px solid #E0CA3C;
    padding: 5px;
    margin: auto;
    display: inline-block;
`;