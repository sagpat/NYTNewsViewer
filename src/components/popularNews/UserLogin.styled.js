import styled from 'styled-components';

export const LoginContainer = styled.div`
    margin: 20% 40%;
`;
export const LoginText = styled.h3`
`;

export const UserName = styled.input`
    padding: 0 2px 0 10px;
    height: 40px;
    width: 84%;
    box-sizing: border-box;
    border: 1px solid;
    margin-bottom: 15px;
`;

export const Password = styled.input`
    padding: 0 2px 0 10px;
    height: 40px;
    width: 84%;
    box-sizing: border-box;
    border: 1px solid;
    margin-bottom: 15px;
`;

export const Button = styled.button`
    padding: 10px 15px;
    display: block;
    margin-left: 60.5%;
    text-align: center;
    font-size: 16px;
    text-decoration: none;
    padding: 15px 20px;
    border: none;
    color: #fff;
    background-color: #000;
`;

export const Divider = styled.hr`
    border-color: #fff;
    boder-style: solid;
    boder-top: none;
    margin: 20px 0;
`;