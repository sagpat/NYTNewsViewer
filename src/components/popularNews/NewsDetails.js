import React from 'react';
import { SummaryParagraph, Section, HeadingTwoAuthor, HeadingTwoDate } from './PopularNews-styled';

const NewsDetails = (props) => {
    return (
        <React.Fragment>
            <SummaryParagraph>{props.summary}</SummaryParagraph>
            <Section>
                <HeadingTwoAuthor>{props.writtenBy ? props.writtenBy : '-'}</HeadingTwoAuthor>
                <HeadingTwoDate>Published Date: {props.publishedDate}</HeadingTwoDate>
            </Section>
        </React.Fragment>
    );
};

export default NewsDetails;
