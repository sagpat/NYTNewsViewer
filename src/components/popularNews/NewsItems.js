import React, { useState } from 'react';
import {
    ListedItem,
    ItemWrapper,
    TitleButton,
    Button,
    Image
} from './PopularNews-styled';
import NewsDetails from './NewsDetails';


const onClickShowCompleteNewsButton = (newsLink) => {
    window.open(newsLink, '_blank');
}

const NewsItems = (props) => {
    const [showNewsDetails, setNewsDetails] = useState(false);
    const [{ 'media-metadata': [{ url }] }] = props.media;

        return (
            <ListedItem >
                <Image src={url} height="100px" width="100px" />
                <ItemWrapper>
                    <TitleButton onClick={ () => setNewsDetails(!showNewsDetails)}>{props.title}</TitleButton>
                    {showNewsDetails &&
                        <NewsDetails
                            summary={props.summary}
                            writtenBy={props.writtenBy}
                            publishedDate={props.publishedDate}
                            section={props.section}
                        />
                    }
                </ItemWrapper>
                <Button onClick={() => onClickShowCompleteNewsButton(props.newsLink)}>></Button>
            </ListedItem>
        );
    }

export default NewsItems;