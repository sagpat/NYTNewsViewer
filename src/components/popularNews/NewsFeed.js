import React from 'react';
import { Heading, UnorderedList } from './NewsFeed-styled';
import NewsItems from './NewsItems';

const NewsFeed = (props) => {
    return (
        <React.Fragment>
            <Heading>NYTimes Most Popular News Feed</Heading>
            {props.popularNewsData.map((data) => {
                return (
                    <UnorderedList key={data.views}>
                        <NewsItems
                            summary={data.abstract}
                            writtenBy={data.byline}
                            publishedDate={data.published_date}
                            newsLink={data.url}
                        />
                    </UnorderedList>
                )
            })

            }
        </React.Fragment>
    );
};

export default NewsFeed;