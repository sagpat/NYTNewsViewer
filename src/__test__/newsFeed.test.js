import React from 'react';
import { shallow } from 'enzyme';
import NewsFeed from '../components/popularNews/NewsFeed';
import { Heading, UnorderedList } from '../components/popularNews/PopularNews-styled';

let wrapper;
const popularNewsData = [
    {
        abstract: 'Test news data',
        byline: '"By MICHAEL D. SHEAR',
        published_date: '23/05/2019',
        url: '"https://www.nytimes.com/2019/06/20/world/middleeast/iran-us-drone.html"',
        media: [
            {
                caption: "“Let’s see what happens,” President Trump said Thursday after Iran shot down an American surveillance drone.",
                copyright: "Erin Schaff/The New York Times"
            }
        ]
    }
];

beforeEach(() => {
    wrapper = shallow(<NewsFeed popularNewsData={popularNewsData} />);
});

describe('NewsFeed rendering', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render one heading', () => {
        expect(wrapper.find(Heading)).toHaveLength(1);
    });

    it('should render one ul element', () => {
        expect(wrapper.find(UnorderedList)).toHaveLength(1);
    });
});

