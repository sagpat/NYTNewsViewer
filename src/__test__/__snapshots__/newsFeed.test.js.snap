// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsFeed rendering renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsFeed
    popularNewsData={
      Array [
        Object {
          "abstract": "Test news data",
          "byline": "\\"By MICHAEL D. SHEAR",
          "media": Array [
            Object {
              "caption": "“Let’s see what happens,” President Trump said Thursday after Iran shot down an American surveillance drone.",
              "copyright": "Erin Schaff/The New York Times",
            },
          ],
          "published_date": "23/05/2019",
          "url": "\\"https://www.nytimes.com/2019/06/20/world/middleeast/iran-us-drone.html\\"",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef(styled.header)>
          NYTimes Most Popular News Feed
        </ForwardRef(styled.header)>,
        Array [
          <ForwardRef(styled.ul)>
            <NewsItems
              media={
                Array [
                  Object {
                    "caption": "“Let’s see what happens,” President Trump said Thursday after Iran shot down an American surveillance drone.",
                    "copyright": "Erin Schaff/The New York Times",
                  },
                ]
              }
              newsLink="\\"https://www.nytimes.com/2019/06/20/world/middleeast/iran-us-drone.html\\""
              publishedDate="23/05/2019"
              summary="Test news data"
              writtenBy="\\"By MICHAEL D. SHEAR"
            />
          </ForwardRef(styled.ul)>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "NYTimes Most Popular News Feed",
        },
        "ref": null,
        "rendered": "NYTimes Most Popular News Feed",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
    text-align: center;
    font-size: 30px;
    padding: 15px;
",
            ],
          },
          "displayName": "styled.header",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "header",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <NewsItems
            media={
              Array [
                Object {
                  "caption": "“Let’s see what happens,” President Trump said Thursday after Iran shot down an American surveillance drone.",
                  "copyright": "Erin Schaff/The New York Times",
                },
              ]
            }
            newsLink="\\"https://www.nytimes.com/2019/06/20/world/middleeast/iran-us-drone.html\\""
            publishedDate="23/05/2019"
            summary="Test news data"
            writtenBy="\\"By MICHAEL D. SHEAR"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "media": Array [
              Object {
                "caption": "“Let’s see what happens,” President Trump said Thursday after Iran shot down an American surveillance drone.",
                "copyright": "Erin Schaff/The New York Times",
              },
            ],
            "newsLink": "\\"https://www.nytimes.com/2019/06/20/world/middleeast/iran-us-drone.html\\"",
            "publishedDate": "23/05/2019",
            "summary": "Test news data",
            "writtenBy": "\\"By MICHAEL D. SHEAR",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
    list-style-type: none;
    margin: 1% 20%;
",
            ],
          },
          "displayName": "styled.ul",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "ul",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef(styled.header)>
            NYTimes Most Popular News Feed
          </ForwardRef(styled.header)>,
          Array [
            <ForwardRef(styled.ul)>
              <NewsItems
                media={
                  Array [
                    Object {
                      "caption": "“Let’s see what happens,” President Trump said Thursday after Iran shot down an American surveillance drone.",
                      "copyright": "Erin Schaff/The New York Times",
                    },
                  ]
                }
                newsLink="\\"https://www.nytimes.com/2019/06/20/world/middleeast/iran-us-drone.html\\""
                publishedDate="23/05/2019"
                summary="Test news data"
                writtenBy="\\"By MICHAEL D. SHEAR"
              />
            </ForwardRef(styled.ul)>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "NYTimes Most Popular News Feed",
          },
          "ref": null,
          "rendered": "NYTimes Most Popular News Feed",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "rules": Array [
                "
    text-align: center;
    font-size: 30px;
    padding: 15px;
",
              ],
            },
            "displayName": "styled.header",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "header",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <NewsItems
              media={
                Array [
                  Object {
                    "caption": "“Let’s see what happens,” President Trump said Thursday after Iran shot down an American surveillance drone.",
                    "copyright": "Erin Schaff/The New York Times",
                  },
                ]
              }
              newsLink="\\"https://www.nytimes.com/2019/06/20/world/middleeast/iran-us-drone.html\\""
              publishedDate="23/05/2019"
              summary="Test news data"
              writtenBy="\\"By MICHAEL D. SHEAR"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "media": Array [
                Object {
                  "caption": "“Let’s see what happens,” President Trump said Thursday after Iran shot down an American surveillance drone.",
                  "copyright": "Erin Schaff/The New York Times",
                },
              ],
              "newsLink": "\\"https://www.nytimes.com/2019/06/20/world/middleeast/iran-us-drone.html\\"",
              "publishedDate": "23/05/2019",
              "summary": "Test news data",
              "writtenBy": "\\"By MICHAEL D. SHEAR",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
    list-style-type: none;
    margin: 1% 20%;
",
              ],
            },
            "displayName": "styled.ul",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "ul",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
